### PR TITLE
[4.0] Hide column default for trashed stages

### PR DESCRIPTION
**Pull request by chmst**
### Summary of Changes
Hide the column "Default" in stages list, when Filter is "trashed", so that trashed items cannot be set as default
trashednew

### Testing Instructions
Go to the stages overview of a workflow. 
Trash one or more items.
Use the filter trashed and see the overview of trashed items.

### Expected result
The column default is now invisible if the filter is "trashed"

### Actual result
The column "Default" is visible and you can set a trashed item to default.
trashed2default

### Documentation Changes Required
no

### CODE REVIEW

File: administrator/components/com_workflow/Model/StageModel.php [hunk 1]
```diff
@@ -261,7 +261,8 @@ public function setDefault($pk, $value = 1)
 
 		if ($table->load(array('id' => $pk)))
 		{
-			if (!$table->published)
+			// Only published stages can be set to defailt
```
Comment (Quy) [inline administrator/components/com_workflow/Model/StageModel.php]: Spelling `defailt`

File: administrator/components/com_workflow/Model/StageModel.php [hunk 2]
```diff
@@ -261,7 +262,8 @@ public function setDefault($pk, $value = 1)
 
 		if ($table->load(array('id' => $pk)))
 		{
-			if (!$table->published)
+			// Only published stages can be set to default
+			if ( $table->published != ContentComponent::CONDITION_PUBLISHED)
```
Comment (Quy) [inline administrator/components/com_workflow/Model/StageModel.php]: Remove space before `$table`

File: administrator/components/com_workflow/Model/StageModel.php [hunk 3]
```diff
@@ -261,7 +262,8 @@ public function setDefault($pk, $value = 1)
 
 		if ($table->load(array('id' => $pk)))
 		{
-			if (!$table->published)
+			// Only published stages can be set to default
+			if ($table->published != ContentComponent::CONDITION_PUBLISHED)
```
Comment (laoneo) [inline administrator/components/com_workflow/Model/StageModel.php:266]: This feels wrong that workflow has a dependency to com_content.

Comment (chmst) [reply administrator/components/com_workflow/Model/StageModel.php:266]: you are right. I'll fix it asap.

**Comment by brianteeman** [comment]
I understand why you are doing this but it feels wrong to show/hide columns when the status changes

We don't do the same for trashed menu items to prevent them being set as default

**Comment by chmst** [comment]
It is true that we cannot have different behaviour. For me it feels better to hide functions everywhere when they are useless. We could do this in menu as well?

**Comment by brianteeman** [comment]
what do you when you set the filter to all?

**Comment by chmst** [comment]
I would hide the icon for trashed items

**Comment by chmst** [comment]
Closing this. I will make a new PR.

**Pull request closed without merging**